app/controllers/api/api_responder.rb

```diff
@@ -0,0 +1,17 @@
+require 'roar/rails/responder'
+
+class Api::ApiResponder < Roar::Rails::Responder
+  def api_behavior
+    raise MissingRenderer.new(format) unless has_renderer?
+
+    if post?
+      display(resource, status: :created)
+    elsif put?
+      display(resource, status: :ok)
+    elsif delete?
+      display(resource, status: :no_content)
```

Comment:
good question.

no, when you tell it to do a 204 (`status: :no_content`), it is smart enough in that display method to know there will be no body.

However, by calling display, it is then smart enough to actually return with errors that get thrown in the action processing. The default before this was to call a `head :no_content`, and that didn't.